src/commands/amend.rs

```diff
@@ -0,0 +1,134 @@
+//! Amend the current commit.
+//!
+//! This command amends the HEAD commit with changes to files
+//! that are already tracked in the repo. Following the amend,
+//! the command performs a restack.
+
+use std::fmt::Write;
+use std::time::SystemTime;
+
+use eyre::Context;
+use itertools::Itertools;
+
+use crate::commands::gc::mark_commit_reachable;
+use crate::commands::restack;
+use crate::core::config::get_restack_preserve_timestamps;
+use crate::core::eventlog::{Event, EventLogDb};
+use crate::git::{AmendFastOptions, FileStatus, GitRunInfo, Repo};
+use crate::opts::MoveOptions;
+use crate::tui::Effects;
+
+/// Amends the existing HEAD commit.
+pub fn amend(
+    effects: &Effects,
+    git_run_info: &GitRunInfo,
+    move_options: &MoveOptions,
+) -> eyre::Result<isize> {
+    let now = SystemTime::now();
+    let repo = Repo::from_current_dir()?;
+    let conn = repo.get_db_conn()?;
+    let mut event_log_db = EventLogDb::new(&conn)?;
+
+    let head_oid = match repo.get_head_info()?.oid {
+        Some(oid) => oid,
+        None => eyre::bail!("No HEAD to amend to"),
+    };
+    let head_commit = repo.find_commit_or_fail(head_oid)?;
+
+    let status = repo.get_status(git_run_info)?;
+    let has_unresolved_conflicts = repo.get_index()?.has_conflicts()
+        || status.iter().any(|s| {
+            s.index_status == FileStatus::Unmerged || s.work_tree_status == FileStatus::Unmerged
+        });
+    if has_unresolved_conflicts {
+        writeln!(
+            effects.get_output_stream(),
+            "Cannot amend, because there are unresolved merge conflicts."
+        )?;
+        return Ok(1);
+    }
+
+    let dirty_index_paths = status
+        .iter()
+        .filter(|s| s.index_status != FileStatus::Unmodified)
+        .map(|s| s.paths())
+        .flatten()
+        .collect_vec();
+    let opts = if !dirty_index_paths.is_empty() {
+        let dirty_working_tree = status
+            .iter()
+            .any(|s| s.work_tree_status != FileStatus::Unmodified);
+        if dirty_working_tree {
+            writeln!(
+                effects.get_output_stream(),
+                "Found staged files in the index. Unstaged changes will not be amended."
+            )?;
+        }
+        AmendFastOptions::FromIndex {
+            paths: dirty_index_paths,
+        }
+    } else {
+        let paths = status
+            .iter()
+            .filter_map(|entry| match entry.work_tree_status {
+                FileStatus::Added
+                | FileStatus::Copied
+                | FileStatus::Deleted
+                | FileStatus::Modified
+                | FileStatus::Renamed => Some(entry.paths()),
+                _ => None,
```

Comment:
Use an exhaustive pattern match here, just in case something changes in the future.